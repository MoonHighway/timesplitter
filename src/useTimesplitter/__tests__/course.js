import { renderHook, act } from "@testing-library/react-hooks";
import "@testing-library/jest-dom";
import { useTimesplitter, TimesplitterProvider } from "..";
import testCourse from "./data/main-sample.json";

it("correctly injects course data", () => {
  const testContent = { title: "Test Course", agenda: [] };
  const wrapper = ({ children }) => (
    <TimesplitterProvider defaultContent={testContent}>
      {children}
    </TimesplitterProvider>
  );
  const { result } = renderHook(() => useTimesplitter(), { wrapper });
  expect(result.current.title).toEqual("Test Course");
});

it("correctly schedules an on time course", () => {
  const wrapper = ({ children }) => (
    <TimesplitterProvider defaultContent={testCourse}>
      {children}
    </TimesplitterProvider>
  );
  const { result } = renderHook(() => useTimesplitter(), { wrapper });
  const sampleDate = new Date("3/28/2021 12:00 PM").getTime();

  expect(result.current.agenda).toMatchSnapshot();
  act(() => result.current.actions.adjust(0, sampleDate));
  expect(result.current.agenda).toMatchSnapshot();
  expect(result.current.time.start).toEqual(sampleDate);

});


