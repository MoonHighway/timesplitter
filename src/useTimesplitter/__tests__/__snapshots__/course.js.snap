// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly schedules an on time course 1`] = `
Array [
  Object {
    "agenda": Array [
      Object {
        "agenda": Array [],
        "difficulty": "beginner",
        "length": 5,
        "required": true,
        "title": "Slides",
        "type": "slides",
      },
      Object {
        "agenda": Array [
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": true,
            "title": "sample step I",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": true,
            "title": "sample step II",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "intermediate",
            "required": false,
            "title": "sample step III",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "advanced",
            "required": false,
            "title": "sample step IV",
            "type": "step",
          },
        ],
        "difficulty": "intermediate",
        "length": 15,
        "required": true,
        "title": "Sample",
        "type": "sample",
      },
      Object {
        "agenda": Array [
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": true,
            "title": "exercise step I",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "intermediate",
            "required": false,
            "title": "exercise step II",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "advanced",
            "required": false,
            "title": "exercise step III",
            "type": "step",
          },
        ],
        "difficulty": "advanced",
        "length": 15,
        "required": true,
        "title": "exercise",
        "type": "exercise",
      },
      Object {
        "agenda": Array [
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": true,
            "title": "lab required step",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "intermediate",
            "required": false,
            "title": "lab bonus step",
            "type": "step",
          },
        ],
        "difficulty": "expert",
        "length": 10,
        "required": true,
        "title": "lab",
        "type": "lab",
      },
    ],
    "difficulty": "beginner",
    "required": true,
    "title": "Timesplitter Topics",
    "type": "section",
  },
  Object {
    "agenda": Array [
      Object {
        "agenda": Array [
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": true,
            "title": "headings and text",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "advanced",
            "required": false,
            "title": "tables",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "expert",
            "required": false,
            "title": "images",
            "type": "step",
          },
        ],
        "difficulty": "beginner",
        "length": 5,
        "required": true,
        "title": "Editing Text",
        "type": "sample",
      },
      Object {
        "agenda": Array [
          Object {
            "difficulty": "intermediate",
            "required": false,
            "title": "react",
          },
          Object {
            "difficulty": "advanced",
            "required": false,
            "title": "command line",
          },
          Object {
            "difficulty": "expert",
            "required": true,
            "title": "css",
          },
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": true,
            "title": "javascript",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": true,
            "title": "graphql",
            "type": "step",
          },
        ],
        "difficulty": "advanced",
        "length": 10,
        "required": true,
        "title": "Code Blocks",
        "type": "exercise",
      },
      Object {
        "agenda": Array [
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": true,
            "title": "Warning",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": true,
            "title": "Error",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": true,
            "title": "Info",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": true,
            "title": "Success",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "intermediate",
            "required": false,
            "title": "Homework",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "advanced",
            "required": false,
            "title": "Checklist",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "intermediate",
            "required": false,
            "title": "Question",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "expert",
            "required": false,
            "title": "line highlighting",
            "type": "step",
          },
        ],
        "difficulty": "intermediate",
        "length": 10,
        "required": true,
        "title": "MDX Components",
        "type": "sample",
      },
    ],
    "difficulty": "intermediate",
    "required": true,
    "title": "Content Editing",
    "type": "section",
  },
  Object {
    "agenda": Array [
      Object {
        "agenda": Array [],
        "difficulty": "expert",
        "length": 10,
        "required": true,
        "title": "Maybe Slides",
        "type": "slides",
      },
      Object {
        "agenda": Array [
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": true,
            "title": "Check this out",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "intermediate",
            "required": true,
            "title": "Then This",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "expert",
            "required": false,
            "title": "Isn't this cool",
            "type": "step",
          },
        ],
        "difficulty": "beginner",
        "length": 10,
        "required": true,
        "title": "Maybe a Sample",
        "type": "sample",
      },
      Object {
        "agenda": Array [
          Object {
            "agenda": Array [],
            "difficulty": "advanced",
            "required": true,
            "title": "let me show you",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "expert",
            "required": true,
            "title": "something cool",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": false,
            "title": "then something else",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "intermediate",
            "required": false,
            "title": "maybe this thing",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "expert",
            "required": false,
            "title": "finally this idea",
            "type": "step",
          },
        ],
        "difficulty": "intermediate",
        "length": 5,
        "required": true,
        "title": "Maybe Another",
        "type": "sample",
      },
      Object {
        "agenda": Array [
          Object {
            "agenda": Array [],
            "difficulty": "expert",
            "required": true,
            "title": "try this step",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": true,
            "title": "then this step",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "intermediate",
            "required": false,
            "title": "maybe we'll do this one",
            "type": "step",
          },
        ],
        "difficulty": "advanced",
        "length": 10,
        "required": true,
        "title": "Maybe an Exercise",
        "type": "exercise",
      },
      Object {
        "agenda": Array [
          Object {
            "agenda": Array [],
            "difficulty": "advanced",
            "required": true,
            "title": "short step",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "expert",
            "required": false,
            "title": "edgecase",
            "type": "step",
          },
        ],
        "difficulty": "expert",
        "length": 5,
        "required": true,
        "title": "Sandwich Sample",
        "type": "sample",
      },
      Object {
        "agenda": Array [
          Object {
            "agenda": Array [],
            "difficulty": "intermediate",
            "required": true,
            "title": "last step for app",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "advanced",
            "required": false,
            "title": "bonus step",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "expert",
            "required": false,
            "title": "bonus feature",
            "type": "step",
          },
        ],
        "difficulty": "beginner",
        "length": 5,
        "required": true,
        "title": "Follow up Exercise",
        "type": "exercise",
      },
      Object {
        "agenda": Array [
          Object {
            "agenda": Array [],
            "difficulty": "advanced",
            "required": true,
            "title": "must complete",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "expert",
            "required": true,
            "title": "hope to complete",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": false,
            "title": "maybe complete",
            "type": "step",
          },
        ],
        "difficulty": "intermediate",
        "length": 5,
        "required": true,
        "title": "Maybe a Lab",
        "type": "lab",
      },
    ],
    "difficulty": "advanced",
    "required": true,
    "title": "Example Section",
    "type": "section",
  },
]
`;

exports[`correctly schedules an on time course 2`] = `
Array [
  Object {
    "agenda": Array [
      Object {
        "agenda": Array [],
        "difficulty": "beginner",
        "length": 5,
        "required": true,
        "time": Object {
          "est": 5,
        },
        "title": "Slides",
        "type": "slides",
      },
      Object {
        "agenda": Array [
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": true,
            "title": "sample step I",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": true,
            "title": "sample step II",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "intermediate",
            "required": false,
            "title": "sample step III",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "advanced",
            "required": false,
            "title": "sample step IV",
            "type": "step",
          },
        ],
        "difficulty": "intermediate",
        "length": 15,
        "required": true,
        "time": Object {
          "est": 15,
        },
        "title": "Sample",
        "type": "sample",
      },
      Object {
        "agenda": Array [
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": true,
            "title": "exercise step I",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "intermediate",
            "required": false,
            "title": "exercise step II",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "advanced",
            "required": false,
            "title": "exercise step III",
            "type": "step",
          },
        ],
        "difficulty": "advanced",
        "length": 15,
        "required": true,
        "time": Object {
          "est": 15,
        },
        "title": "exercise",
        "type": "exercise",
      },
      Object {
        "agenda": Array [
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": true,
            "title": "lab required step",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "intermediate",
            "required": false,
            "title": "lab bonus step",
            "type": "step",
          },
        ],
        "difficulty": "expert",
        "length": 10,
        "required": true,
        "time": Object {
          "est": 10,
        },
        "title": "lab",
        "type": "lab",
      },
    ],
    "difficulty": "beginner",
    "required": true,
    "time": Object {
      "est": 45,
    },
    "title": "Timesplitter Topics",
    "type": "section",
  },
  Object {
    "agenda": Array [
      Object {
        "agenda": Array [
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": true,
            "title": "headings and text",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "advanced",
            "required": false,
            "title": "tables",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "expert",
            "required": false,
            "title": "images",
            "type": "step",
          },
        ],
        "difficulty": "beginner",
        "length": 5,
        "required": true,
        "time": Object {
          "est": 5,
        },
        "title": "Editing Text",
        "type": "sample",
      },
      Object {
        "agenda": Array [
          Object {
            "agenda": Array [],
            "difficulty": "intermediate",
            "required": false,
            "title": "react",
          },
          Object {
            "agenda": Array [],
            "difficulty": "advanced",
            "required": false,
            "title": "command line",
          },
          Object {
            "agenda": Array [],
            "difficulty": "expert",
            "required": true,
            "title": "css",
          },
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": true,
            "title": "javascript",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": true,
            "title": "graphql",
            "type": "step",
          },
        ],
        "difficulty": "advanced",
        "length": 10,
        "required": true,
        "time": Object {
          "est": 10,
        },
        "title": "Code Blocks",
        "type": "exercise",
      },
      Object {
        "agenda": Array [
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": true,
            "title": "Warning",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": true,
            "title": "Error",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": true,
            "title": "Info",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": true,
            "title": "Success",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "intermediate",
            "required": false,
            "title": "Homework",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "advanced",
            "required": false,
            "title": "Checklist",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "intermediate",
            "required": false,
            "title": "Question",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "expert",
            "required": false,
            "title": "line highlighting",
            "type": "step",
          },
        ],
        "difficulty": "intermediate",
        "length": 10,
        "required": true,
        "time": Object {
          "est": 10,
        },
        "title": "MDX Components",
        "type": "sample",
      },
    ],
    "difficulty": "intermediate",
    "required": true,
    "time": Object {
      "est": 25,
    },
    "title": "Content Editing",
    "type": "section",
  },
  Object {
    "agenda": Array [
      Object {
        "agenda": Array [],
        "difficulty": "expert",
        "length": 10,
        "required": true,
        "time": Object {
          "est": 10,
        },
        "title": "Maybe Slides",
        "type": "slides",
      },
      Object {
        "agenda": Array [
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": true,
            "title": "Check this out",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "intermediate",
            "required": true,
            "title": "Then This",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "expert",
            "required": false,
            "title": "Isn't this cool",
            "type": "step",
          },
        ],
        "difficulty": "beginner",
        "length": 10,
        "required": true,
        "time": Object {
          "est": 10,
        },
        "title": "Maybe a Sample",
        "type": "sample",
      },
      Object {
        "agenda": Array [
          Object {
            "agenda": Array [],
            "difficulty": "advanced",
            "required": true,
            "title": "let me show you",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "expert",
            "required": true,
            "title": "something cool",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": false,
            "title": "then something else",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "intermediate",
            "required": false,
            "title": "maybe this thing",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "expert",
            "required": false,
            "title": "finally this idea",
            "type": "step",
          },
        ],
        "difficulty": "intermediate",
        "length": 5,
        "required": true,
        "time": Object {
          "est": 5,
        },
        "title": "Maybe Another",
        "type": "sample",
      },
      Object {
        "agenda": Array [
          Object {
            "agenda": Array [],
            "difficulty": "expert",
            "required": true,
            "title": "try this step",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": true,
            "title": "then this step",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "intermediate",
            "required": false,
            "title": "maybe we'll do this one",
            "type": "step",
          },
        ],
        "difficulty": "advanced",
        "length": 10,
        "required": true,
        "time": Object {
          "est": 10,
        },
        "title": "Maybe an Exercise",
        "type": "exercise",
      },
      Object {
        "agenda": Array [
          Object {
            "agenda": Array [],
            "difficulty": "advanced",
            "required": true,
            "title": "short step",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "expert",
            "required": false,
            "title": "edgecase",
            "type": "step",
          },
        ],
        "difficulty": "expert",
        "length": 5,
        "required": true,
        "time": Object {
          "est": 5,
        },
        "title": "Sandwich Sample",
        "type": "sample",
      },
      Object {
        "agenda": Array [
          Object {
            "agenda": Array [],
            "difficulty": "intermediate",
            "required": true,
            "title": "last step for app",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "advanced",
            "required": false,
            "title": "bonus step",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "expert",
            "required": false,
            "title": "bonus feature",
            "type": "step",
          },
        ],
        "difficulty": "beginner",
        "length": 5,
        "required": true,
        "time": Object {
          "est": 5,
        },
        "title": "Follow up Exercise",
        "type": "exercise",
      },
      Object {
        "agenda": Array [
          Object {
            "agenda": Array [],
            "difficulty": "advanced",
            "required": true,
            "title": "must complete",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "expert",
            "required": true,
            "title": "hope to complete",
            "type": "step",
          },
          Object {
            "agenda": Array [],
            "difficulty": "beginner",
            "required": false,
            "title": "maybe complete",
            "type": "step",
          },
        ],
        "difficulty": "intermediate",
        "length": 5,
        "required": true,
        "time": Object {
          "est": 5,
        },
        "title": "Maybe a Lab",
        "type": "lab",
      },
    ],
    "difficulty": "advanced",
    "required": true,
    "time": Object {
      "est": 50,
    },
    "title": "Example Section",
    "type": "section",
  },
]
`;
